src/common/meta/src/key/datanode_table.rs

```diff
@@ -165,11 +169,21 @@ impl DatanodeTableManager {
         engine: &str,
         region_storage_path: &str,
         region_options: HashMap<String, String>,
+        region_wal_options: HashMap<RegionNumber, String>,
         distribution: RegionDistribution,
     ) -> Result<Txn> {
         let txns = distribution
             .into_iter()
             .map(|(datanode_id, regions)| {
+                let filtered_region_wal_options = regions
+                    .iter()
+                    .filter_map(|region_number| {
+                        region_wal_options
+                            .get(region_number)
+                            .map(|wal_options| (region_number.to_string(), wal_options.clone()))
+                    })
+                    .collect();
```

Comment:
I'm not sure, but I've seen these codes several times. Should we move it to a separate method?

Reply:
I've checked and there're two usages of such a code snippet. However, they have a little difference. 
Will reuse the code with a utility function if necessary in the future.